### ISSUE TITLE
Non-free network dependency?

### ISSUE THREAD

**Issue by IzzySoft** [breezy-weather/breezy-weather#1206]
### Steps to reproduce

Since its last release, the app contains the [Google API Client Libraries](https://developers.google.com/api-client-library/java/) – which trigger the `NonFreeNet` anti-feature. I don't know what that would be needed for, but maybe you could remove it again? (I'm talking about the "Standard" build here which is made available at IzzyOnDroid) Until then, the mentioned anti-feature will be raised. Thanks in advance for your help!

### Expected behavior

that API not being present (probably not needed for weather, or?)

### Actual behavior

isPresent()

### Weather source used

n/a

### Breezy Weather version

n/a

### Android version

n/a

### Device

n/a

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [X] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [X] I have updated the app to **[version 5.2.6 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)** and I can reproduce the issue in this version.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
I don't think we have any non-free component?

We have Google Maps Utils components, but only the open source part: https://github.com/breezy-weather/breezy-weather/tree/v5.2.6/maps-utils

This was not added in v5.2.6, so let me know if it was pulled by the update of one of our dependencies maybe?

**Comment by IzzySoft** [comment]
I cannot tell what pulled it in, I only see it is there. You could try `gradle :app:dependencies` to find out, it should list a tree. I wouldn't wonder if Google Maps Utils pulled that in – though it wasn't added in this release, maybe it was updated?

OTOH, maybe this from the changelog?

> Add an option to periodically check for app updates in the background. Available from Settings > Background updates. Disabled by default.

Does that maybe check at PlayStore even?

> In the freenet flavor where non-free networks are not allowed

:thinking:

**Comment by papjul** [comment]
Google Maps Utils itself is just an Apache 2.0 lib. We forked it to remove the hard dependency to any proprietary component. This lib is now maintained in our project and anyone can use it and check that it has no proprietary dependency: https://github.com/breezy-weather/breezy-weather/blob/main/maps-utils/build.gradle.kts

Once again, this lib was added in v5.0.3-alpha, so it’s unrelated to anything added in v5.2.6.

Regarding the update check, it just pulls the GitHub API, a non-free network, not the Play Store where our app is not distributed.

> I cannot tell what pulled it in, I only see it is there.

What do you see *exactly* that you didn’t see in v5.2.5?

> You could try `gradle :app:dependencies` to find out

Doesn’t seem to work

**Comment by IzzySoft** [comment]
> What do you see exactly that you didn’t see in v5.2.5?

![image](https://github.com/user-attachments/assets/0aa4921e-914f-489e-a903-dbee14607d17)

But I guess I found the culprit – and it was not at your end, apologies. This seems to be a reported only by LibRadar. When disabling that in my scanner package, the entry was gone – the signatures of my own engine didn't detect it, nor could I find it in the smali code. But then, there's a lot of obfuscation – so I'm not sure if it was a false positive, or LibRadar indeed detected the presence of that library by some bytecode.

In dubio pro reo: I manually removed the entry from the database and the antifeature from the app. Would still be interesting what suddenly triggered it. That would be the first "false positive" I had in the ~8 years I'm using that scanner.

And what do you mean by `gradle :app:dependencies` doesn't work? Crashes, shows no results at all, or just not this dependency? In the latter case, it could also be in one of the modules, like `gradle :maps-utils:dependencies`. Just want to make sure we didn't miss something here. Also see: [Using Gradle to find dependency tree](https://stackoverflow.com/questions/21645071/using-gradle-to-find-dependency-tree) which mentions this.

Thanks for your help!

**Comment by papjul** [comment]
LibRadar source code mentions only once "Google API Client Libraries", and it refers to `com.google.api.client` namespace.

Opening a non-obfuscated version of Breezy Weather in Jadx show no matching result:
![Screenshot_20240719_162539](https://github.com/user-attachments/assets/4a7c8c02-d7df-4abe-9d0d-a1ece8a5a937)

> Would still be interesting what suddenly triggered it.

Our "maps-utils" lib fork was last modified 2 months ago, and even then, it was no significant change. Did this false positive happen with other projects or is it the only project suddenly triggering that?

Between v5.2.5 and v5.2.6, no Google library was updated, except Gradle v8.9 and Android Gradle Plugin from v8.5.0 to v8.5.1, but they should not have any link with the `com.google` namespace.

Since it's able to detect it obfuscated from what I understand, could it be something that it thinks match the signature of Google Play libraries? ~~For example, to remove the proprietary dependency, we had to create a `LatLng` class, which is just a simple class with a `longitude` and `latitude`. Could the lib be looking for a `LatLng` class to detect a proprietary lib, and given how simple this class, no wonder it ends up being a false positive. But once again, nothing was changed since v5.2.5 with this class.~~ Nevermind, the `LatLng` class doesn't belong to the `com.google.api.client` namespace, so it's not that.

> And what do you mean by `gradle :app:dependencies` doesn't work?

Must have been a typo on my side, works now. I have a 32K lines file, but nothing matches "com.google.api.client" namespace.

**Comment by IzzySoft** [comment]
> Do you, by any chance, have a verbose log of what it detected? Like the obfuscated path to the smali file it thinks is Google Api Client Lib.

No, unfortunately that output is not produced here by LibRadar. I have plans to "sunset" it and replace it by something that's still maintained; guess I was one of the last contributors (updating details to the signatures like library names and source URLs) of the "stand-alone" V1 branch there as there were no updates for years except for mine…

> Must have been a typo on my side, works now. I have a 32K lines file, but nothing matches "com.google.api.client" namespace.

Thanks a lot – and apologies for the noise! As pointed out, this was the first (and only) false positive I can remember having had, so I simply relied on the output as it showed to be reliable for many years.

Closing the issue then as it's resolved.
